api/src/main/java/org/pmiops/workbench/actionaudit/targetproperties/ProfileTargetProperty.kt

```diff
@@ -0,0 +1,24 @@
+package org.pmiops.workbench.actionaudit.targetproperties
+
+import org.pmiops.workbench.model.Profile
+
+// N.B. entries will rarely be referred to by name, but are accessed via values(). I.e. they are
+// not safe to remove, even if an IDE indicates otherwise.
+enum class ProfileTargetProperty
+constructor(override val propertyName: String, override val extractor: (Profile) -> String?) : TargetProperty<Profile> {
+    USER_NAME("user_name", Profile::getUsername),
+    CONTACT_EMAIL("contact_email", Profile::getContactEmail),
+    DATA_ACCESS_LEVEL("data_access_level", { it.dataAccessLevel?.toString() }),
```

Comment:
It might help to think of `it` as an operator and not as a variable.

I think a part of Kotlin's philosophy is that you have a rock-solid type system, so you can rest assured there aren't bogus casts happening when you pass things around. In this case, these are all functions or lambdas that take a profile and return a string or null, and putting `profile -> profile` on every line makes readability worse IMHO.